test/jest_integration/saml_auth.tests.ts

```diff
@@ -0,0 +1,278 @@
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    ).then(value => {
+      Promise.resolve(value);
+    }).catch(value => {
+      Promise.resolve(value);
+    });
+    console.log("Remove the Role Mapping");
+    await wreck.patch(
+      'https://localhost:9200/_plugins/_security/api/rolesmapping/all_access',
+      {
+        payload: [{
+          "op": "remove",
+          "path": "/users",
+          "users": ["saml.jackson@example.com"]
+        }],
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    ).then(value => {
+      Promise.resolve(value);
+    }).catch(value => {
+      Promise.resolve(value);
+    });
+    console.log("Remove the Security Config");
+     await wreck.patch(
+      'https://localhost:9200/_plugins/_security/api/securityconfig',
+      {
+        payload: [{
+          "op": "remove",
+          "path": "/config/dynamic/authc/saml_auth_domain",
+        }],
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    ).then(value => {
+       Promise.resolve(value);
+     }).catch(value => {
+       Promise.resolve(value);
+     });
+     //   .then(value => {
+     //
+     // }).catch(reason => {
+     //
+     // });
+    // shutdown OpenSearchDashboards server
+    await root.shutdown();
+    await driver.quit();
+
+  });
+  // 1 Integ Test for first time log in.
+  // 1 Integ Test for Log into Dashboard with Hash
+  // 1 Integ Test for logging into dev console
+  // 1 Integ Test to test Cookie expiry
+  it('Login to app/opensearch_dashboards_overview#/ when SAML is enabled', async () => {
+    await driver.get('http://localhost:5601/app/opensearch_dashboards_overview#/');
+    await driver.findElement(By.id('btn-sign-in')).click();
+    // await driver.wait(until.urlContains('/app/home'));
+    // await driver.wait(until.titleIs('Home - OpenSearch Dashboards'));
+    // let currUrl = await driver.getCurrentUrl();
+    // console.log(currUrl);
+
+    await driver
+      .wait(
+        until.elementsLocated(
+          By.xpath("//*[@id=\"osdOverviewPageHeader__title\"]")
+        ),
+        10000
+      );
+
+    let cookie = await driver.manage().getCookies();
+    console.log(cookie.length);
+    console.log(cookie[1]);
+    expect(cookie.length).toEqual(2);
+    await driver.manage().deleteAllCookies();
+  });
+
+  it('Login to app/dev_tools#/console when SAML is enabled', async () => {
+    await driver.get('http://localhost:5601/app/dev_tools#/console');
+    await driver.findElement(By.id('btn-sign-in')).click();
+
+    await driver
+      .wait(
+        until.elementsLocated(
+          By.xpath("/html/body/div[1]/div/div/div/div[2]/div/main/div[1]/span/button/span")
```

Comment:
Is there a way this can be found by id that would be cleaner?

Reply:
sure, I will need to look at the DOM to see for ID's